R/dm.R

```diff
@@ -116,13 +119,15 @@ new_dm <- function(tables = list()) {
 
   def <-
     tibble(table, data, segment = NA_character_, display = NA_character_) %>%
-    left_join(pks, by = "table") %>%
-    left_join(fks, by = "table") %>%
-    left_join(filters, by = "table") %>%
+    left_join(pks_df, by = "table") %>%
+    mutate(pks = as_list_of(map(pks, `%||%`, new_pk()), .ptype = new_pk())) %>%
+    left_join(fks_df, by = "table") %>%
+    mutate(fks = as_list_of(map(fks, `%||%`, new_fk()), .ptype = new_fk())) %>%
+    mutate(filters = list_of(new_filter())) %>%
     left_join(zoom, by = "table") %>%
     left_join(col_tracker_zoom, by = "table")
 
-  new_dm3(def, validate = FALSE)
+  new_dm3(def, validate = validate)
 }
 
 new_dm3 <- function(def, zoomed = FALSE, validate = TRUE) {
```

Comment:
Why don't we enable validation now?

Reply:
It's slow. #597 adds a workflow that enables this. It currently fails, will succeed after we merge this PR.